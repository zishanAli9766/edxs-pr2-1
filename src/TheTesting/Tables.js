import React from "react";

const Tables = () =>{
    return(
        <table className="custom-border">
<thead>
    <tr>
        <th>Name</th>
        <th>Email</th>
    </tr>
</thead>
<tbody>
    <tr>
        <td>First</td>
        <td>Na</td>
    </tr>
</tbody>
        </table>
    )
}

export default Tables