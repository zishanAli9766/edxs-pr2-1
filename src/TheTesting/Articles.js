import React,{Component} from "react";
import Tables from "./Tables";
//Simulate & Dive
class Articles extends Component{
    constructor(props){
        super(props);
        this.state = {
            count : 0,
        }
    }
    render(){
        const {count} = this.state;
        return(
            <div>
<div className={`clicks-${count}`}>
{count}  Clicks
</div>
<button onClick={()=>{this.setState({count: count +1 })}}>
    Increment
</button>
<h1>Article Details</h1>
<Tables />
            </div>
        )
    }
}

export default Articles