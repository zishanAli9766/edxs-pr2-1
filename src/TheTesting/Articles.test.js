import Enzyme, { render, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Articles from './Articles';
import Tables from './Tables';
Enzyme.configure({ adapter: new Adapter() });


describe('Article page',()=>{
    it('click Count',()=>{
        let words = shallow(<Articles />)
        // console.log(words.debug());
        expect(words.find('.clicks-0').length).toEqual(1)
        words.find('button').simulate('click')
        expect(words.find('.clicks-1').length).toEqual(1)
    })
    it('click Count incre 2',()=>{
        let words = shallow(<Articles />)
        // console.log(words.debug());
        expect(words.find('.clicks-0').length).toEqual(1)
        words.find('button').simulate('click')
        words.find('button').simulate('click')
        expect(words.find('.clicks-2').length).toEqual(1)
    })
    it('Dive in Table',()=>{
        let variables = shallow(<Articles />)
        expect(variables.find(Tables).dive().find('.custom-border').length).toBe(1)
    })
    it('SnapShot 1',()=>{
        let variables = shallow(<Tables/>)
        expect(variables).toMatchSnapshot();
    })
})