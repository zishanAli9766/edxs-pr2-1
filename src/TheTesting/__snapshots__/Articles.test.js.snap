// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article page SnapShot 1 1`] = `ShallowWrapper {}`;
